src/python/pants/option/options_bootstrapper.py

```diff
@@ -3,255 +3,144 @@
 
 from __future__ import annotations
 
-import itertools
 import os
-import warnings
 from dataclasses import dataclass
-from pathlib import Path
 from typing import TYPE_CHECKING, Iterable, Mapping, Sequence
 
-from pants.base.build_environment import get_buildroot, get_default_pants_config_file, pants_version
+from pants.base.build_environment import get_buildroot, pants_version
 from pants.base.exceptions import BuildConfigurationError
 from pants.engine.unions import UnionMembership
-from pants.option.config import Config
-from pants.option.custom_types import ListValueComponent
 from pants.option.global_options import BootstrapOptions, GlobalOptions
 from pants.option.option_types import collect_options_info
 from pants.option.options import Options
 from pants.option.scope import GLOBAL_SCOPE, ScopeInfo
-from pants.option.subsystem import Subsystem
-from pants.util.dirutil import read_file
+from pants.util.frozendict import FrozenDict
 from pants.util.memo import memoized_method, memoized_property
-from pants.util.ordered_set import FrozenOrderedSet
-from pants.util.strutil import ensure_text, softwrap
+from pants.util.strutil import softwrap
 
 if TYPE_CHECKING:
     from pants.build_graph.build_configuration import BuildConfiguration
 
 
+# TODO: Rebrand this. It isn't actually about "bootstrapping" any more (and the term
+#  "bootstrap options" now means just "options needed to create a Scheduler").
+
+
 @dataclass(frozen=True)
 class OptionsBootstrapper:
-    """Holds the result of the first stage of options parsing, and assists with parsing full
-    options."""
+    """Creates Options instances with appropriately registered options."""
 
-    env_tuples: tuple[tuple[str, str], ...]
-    bootstrap_args: tuple[str, ...]
     args: tuple[str, ...]
-    config: Config
+    env: FrozenDict[str, str]
+    allow_pantsrc: bool
 
     def __repr__(self) -> str:
-        env = {pair[0]: pair[1] for pair in self.env_tuples}
+        env = {pair[0]: pair[1] for pair in self.env.items()}
         # Bootstrap args are included in `args`. We also drop the first argument, which is the path
         # to `pants_loader.py`.
         args = list(self.args[1:])
-        return f"OptionsBootstrapper(args={args}, env={env}, config={self.config})"
-
-    @staticmethod
-    def get_config_file_paths(env, args) -> list[str]:
```

Comment:
This chicken-and-egg problem no longer exists. Or rather, it is handled much more elegantly in the Rust code. So this entire special-casing monstrosity can go.